packages/shared/lib/services/connection.service.ts

```diff
@@ -522,26 +524,42 @@ class ConnectionService {
         await trx.from<StoredConnection>(`_nango_connections`).whereIn('id', ids).andWhere({ deleted: false }).update({ metadata });
     }
 
-    public async replaceConnectionConfig(connection: Connection, config: ConnectionConfig) {
-        await db.knex
+    public async replaceConnectionConfig(connection: Connection, config: ConnectionConfig, trx?: Knex.Transaction) {
+        const query = db.knex
             .from<StoredConnection>(`_nango_connections`)
             .where({ id: connection.id as number, deleted: false })
             .update({ connection_config: config });
+
+        if (trx) {
+            await query.transacting(trx);
+        } else {
+            await query;
+        }
     }
 
-    public async updateMetadata(connections: Connection[], metadata: Metadata): Promise<void> {
-        await db.knex.transaction(async (trx) => {
+    public async updateMetadata(connections: Connection[], metadata: Metadata, trx?: Knex.Transaction): Promise<void> {
```

Comment:
The functionality here stays the same, but needed to add an optional transaction parameter so this can be batched with updating the connection config.